marine/marine.py

```diff
@@ -154,3 +175,37 @@ def _add_or_get_filter(
 
     def __del__(self):
         self._marine.destroy_marine()
+
+    @classmethod
+    def _expand_macros(
+        cls, fields: List[str], macros: Optional[Dict[str, List[str]]]
+    ) -> List[str]:
+        if not macros:
+            return fields
+
+        return [
```

Comment:
I think we should consider surrounding that with `list({...})` instead, to avoid dissecting the same field several times in case of overlapping macros.
Not sure if that's a common case (I guess it's not rare but probably not that significant), but if that optimization is cheap there's no reason to not implement it.